### ISSUE TITLE
Backend container crashes randomly and does not restart

### ISSUE THREAD

**Issue by lazavikmaria** [blockscout/blockscout#9040]
### Description

Hi team,
I use docker-compose to run Blockscout, but the app inside the backend container sometimes stops working and the container does not restart automatically.

Logs:
```
=CRASH REPORT==== 21-Dec-2023::06:46:07.133206 ===
  crasher:
    initial call: erlang:apply/2
    pid: <0.16435.23>
    registered_name: []
    exception exit: {noproc,
                        {'Elixir.GenServer',call,
                            [<0.15922.23>,
                             {push_back,
                                 #{'__struct__' => 'Elixir.Range',
                                   first => 1461355,last => 1461350,
                                   step => -1}},
                             5000]}}
      in function  'Elixir.GenServer':call/3 (lib/gen_server.ex, line 1038)
      in call from 'Elixir.Indexer.Block.Catchup.Fetcher':push_back/2 (lib/indexer/block/catchup/fetcher.ex, line 274)
      in call from 'Elixir.Indexer.Block.Catchup.Fetcher':fetch_and_import_range_from_sequence/3 (lib/indexer/block/catchup/fetcher.ex, line 231)
      in call from 'Elixir.Task.Supervised':invoke_mfa/2 (lib/task/supervised.ex, line 89)
      in call from 'Elixir.Task.Supervised':reply/4 (lib/task/supervised.ex, line 34)
    ancestors: ['Elixir.Indexer.Block.Catchup.TaskSupervisor',
                  'Elixir.Indexer.Block.Catchup.Supervisor',
                  'Elixir.Indexer.Supervisor','Elixir.Indexer.Application',
                  <0.4623.0>]
    message_queue_len: 2
    messages: [{'EXIT',<0.15873.23>,shutdown},
                  {'EXIT',<0.15924.23>,shutdown}]
    links: []
    dictionary: [{rand_seed,
                      {#{bits => 58,jump => #Fun<rand.3.92093067>,
                         next => #Fun<rand.0.92093067>,type => exsss,
                         uniform => #Fun<rand.1.92093067>,
                         uniform_n => #Fun<rand.2.92093067>},
                       [138371433664540450|60714290806215131]}},
                  {'$logger_metadata$',
                      #{fetcher => block_catchup,
                        first_block_number => 1461355,
                        last_block_number => 1461350}},
                  {'$callers',[<0.15874.23>,<0.15871.23>]}]
    trap_exit: true
    status: running
    heap_size: 6772
    stack_size: 28
    reductions: 160559
  neighbours:

=SUPERVISOR REPORT==== 21-Dec-2023::06:46:07.135497 ===
    supervisor: {local,'Elixir.Indexer.Block.Catchup.TaskSupervisor'}
    errorContext: shutdown_error
    reason: {noproc,
                {'Elixir.GenServer',call,
                    [<0.15922.23>,
                     {push_back,
                         #{'__struct__' => 'Elixir.Range',first => 1461254,
                           last => 1461254,step => 1}},
                     5000]}}
    offender: [{pid,<0.16420.23>},
               {id,undefined},
               {mfargs,{'Elixir.Task.Supervised',start_link,undefined}},
               {restart_type,temporary},
               {shutdown,300000},
               {child_type,worker}]
=SUPERVISOR REPORT==== 21-Dec-2023::06:46:07.135708 ===
    supervisor: {local,'Elixir.Indexer.Block.Catchup.TaskSupervisor'}
    errorContext: shutdown_error
    reason: {noproc,
                {'Elixir.GenServer',call,
                    [<0.15922.23>,
                     {push_back,
                         #{'__struct__' => 'Elixir.Range',first => 1461355,
                           last => 1461350,step => -1}},
                     5000]}}
    offender: [{pid,<0.16435.23>},
               {id,undefined},
               {mfargs,{'Elixir.Task.Supervised',start_link,undefined}},
               {restart_type,temporary},
               {shutdown,300000},
               {child_type,worker}]
=SUPERVISOR REPORT==== 21-Dec-2023::06:46:07.136452 ===
    supervisor: {local,'Elixir.Indexer.Block.Catchup.Supervisor'}
    errorContext: shutdown_error
    reason: {#{'__exception__' => true,
               '__struct__' => 'Elixir.Postgrex.Error',connection_id => 56309,
               message => nil,
               postgres =>
                   #{code => query_canceled,file => <<"postgres.c">>,
                     line => <<"3342">>,
                     message => <<"canceling statement due to user request">>,
                     pg_code => <<"57014">>,
                     routine => <<"ProcessInterrupts">>,
                     severity => <<"ERROR">>,unknown => <<"ERROR">>},
               query => nil},
             [{'Elixir.Ecto.Adapters.SQL',raise_sql_call_error,1,
                  [{file,"lib/ecto/adapters/sql.ex"},
                   {line,1054},
                   {error_info,#{module => 'Elixir.Exception'}}]},
              {'Elixir.Ecto.Adapters.SQL',execute,6,
                  [{file,"lib/ecto/adapters/sql.ex"},{line,952}]},
              {'Elixir.Ecto.Repo.Queryable',execute,4,
                  [{file,"lib/ecto/repo/queryable.ex"},{line,232}]},
              {'Elixir.Explorer.Utility.MissingBlockRange',
                  delete_intersecting_ranges,0,
                  [{file,"lib/explorer/utility/missing_block_range.ex"},
                   {line,178}]},
              {'Elixir.Explorer.Utility.MissingBlockRange',
                  delete_and_merge_ranges,0,
                  [{file,"lib/explorer/utility/missing_block_range.ex"},
                   {line,156}]},
              {'Elixir.Explorer.Utility.MissingBlockRange',
                  sanitize_missing_block_ranges,0,
                  [{file,"lib/explorer/utility/missing_block_range.ex"},
                   {line,150}]},
              {'Elixir.Indexer.Block.Catchup.MissingRangesCollector',
                  default_init,0,
                  [{file,
                       "lib/indexer/block/catchup/missing_ranges_collector.ex"},
                   {line,48}]},
              {'Elixir.Indexer.Block.Catchup.MissingRangesCollector',
                  handle_continue,2,
                  [{file,
                       "lib/indexer/block/catchup/missing_ranges_collector.ex"},
                   {line,30}]}]}
    offender: [{pid,<0.15870.23>},
               {id,'Elixir.Indexer.Block.Catchup.MissingRangesCollector'},
               {mfargs,
                   {'Elixir.Indexer.Block.Catchup.MissingRangesCollector',
                       start_link,
                       [[]]}},
               {restart_type,permanent},
               {significant,false},
               {shutdown,5000},
               {child_type,worker}]
```

### Type of the installation

Docker-compose

### Type of the JSON RPC archive node

fendermint

### Backend version

v5.3.3

### Elixir & Erlang/OTP versions

Elixir 1.14.5 (compiled with Erlang/OTP 24)

**Comment by maciejwitowski** [comment]
@lazavikmaria 
Follow up questions from the blockscout team
> - which docker image was used exactly?
> - and this ev var was set ETHEREUM_JSONRPC_TRANSPORT=ipc , right?

**Comment by lazavikmaria** [comment]
docker image is `blockscout/blockscout:latest `
and env `ETHEREUM_JSONRPC_TRANSPORT=http`

**Comment by vbaranov** [comment]
@lazavikmaria we recently published 6.0.0. Could you please update latest tag and let me know if the issue with crashing still persist? If it is present, could you please send the full log from the run?

**Comment by vbaranov** [comment]
Closing due to no response from the author.
